src/components/dialog.js

```diff
@@ -0,0 +1,102 @@
+(() => ({
+  name: 'Dialog',
+  type: 'CONTAINER_COMPONENT',
+  allowedTypes: ['BODY_COMPONENT', 'CONTENT_COMPONENT'],
+  orientation: 'HORIZONTAL',
+  jsx: (() => {
+    const { env } = B;
+    const { Dialog } = window.MaterialUI.Core;
+
+    const isEmpty = children.length === 0;
+    const isDev = env === 'dev';
+    const isPristine = isDev && isEmpty;
+    const isContentVisible = isDev && options.isVisibleInDev;
+
+    const classNames = isContentVisible
+      ? [
+          classes.content,
+          isEmpty ? classes.empty : '',
+          isPristine ? classes.pristine : '',
+        ].join(' ')
+      : '';
+
+    const [open, setOpen] = React.useState(false);
+    const devContent = isContentVisible && !isEmpty ? children : 'Dialog';
+
+    useEffect(() => {
+      B.defineFunction('handleDialogOpen', e => {
+        e.preventDefault();
+
+        setOpen(true);
+      });
+
+      B.defineFunction('handleDialogClose', e => {
+        e.preventDefault();
+
+        setOpen(false);
+      });
+    }, []);
+
+    return (
+      <div className={classNames}>
+        {isDev ? devContent : ''}
+
+        <Dialog open={open} onClose={() => setOpen(false)}>
+          <div className={classes.dialog}>{children}</div>
+        </Dialog>
+      </div>
+    );
+  })(),
+  styles: B => theme => {
+    const style = new B.Styling(theme);
+    const getSpacing = (idx, device = 'Mobile') =>
+      idx === '0' ? '0rem' : style.getSpacing(idx, device);
+    return {
+      dialog: {
+        backgroundColor: 'white',
+        margin: '0 auto',
+        padding: '1.5rem',
+        overflowWrap: 'break-word',
+      },
+      content: {
+        flexGrow: 1,
+        flexBasis: '100%',
+        paddingTop: getSpacing('M'),
+        paddingRight: getSpacing('M'),
+        paddingBottom: getSpacing('M'),
+        paddingLeft: getSpacing('M'),
+      },
+      [`@media ${B.mediaMinWidth(768)}`]: {
+        content: {
+          paddingTop: getSpacing('M', 'Portrait'),
+        },
+      },
+      [`@media ${B.mediaMinWidth(1024)}`]: {
+        content: {
+          padding: getSpacing('M', 'Landscape'),
+        },
+      },
+      [`@media ${B.mediaMinWidth(1200)}`]: {
+        content: {
+          padding: getSpacing('M', 'Desktop'),
+        },
+      },
+      empty: {
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: () => 'center',
+        minHeight: '4rem',
+        height: '100%',
```

Comment:
This height of 100% makes a column show a scrollbar. Please think of another solution. I guess the min height of 4rem will do. The empty class name is meant for showing users they dragged a container but didn't drag any children in it.